File: 0x0 initial-utilities/wcat/wcat.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]){
  // loop on all input files
    for (int i = 1; i < argc; i++){ // starts from 1 as argv[0] has program name
        char *file = argv[i]; // store file name into string
        FILE *fpntr = fopen(file, "r"); // open file returns a pointer

        if (fpntr == NULL){ // doing some validation
            printf("wcat: cannot open file\n");
            exit(1);
        }

        char buf[256]; // create a buffer to read data
        // there are multple methods to read data (fgets, fgetc, getline, ..)
        while (fgets(buf, sizeof(buf), fpntr) != NULL){
            printf("%s", buf);
        }
        fclose(fpntr);
    }
    return 0;
}

File: 0x0 initial-utilities/wcat/wcat.go
package main

import (
	"fmt"
	"os"
)

func main(){
	files := os.Args[1:]
	
	for _, file := range files {
		data, err := os.ReadFile(file)
	
		if err != nil{
			fmt.Printf("%e", err)
		}

		fmt.Printf("%v", string(data))
	}

}
